auraed/src/runtime/cell_name.rs

```diff
@@ -25,6 +30,23 @@ impl validation::ValidatedField<String> for CellName {
             parent_name,
         )?;
 
+        Ok(input)
+    }
+
+    pub fn into_inner(self) -> String {
+        self.0
+    }
+}
+
+impl ValidatedField<String> for CellName {
+    fn validate(
+        input: Option<String>,
+        field_name: &str,
+        parent_name: Option<&str>,
+    ) -> Result<Self, ValidationError> {
+        let input =
+            validation::required_not_empty(input, field_name, parent_name)?;
+
```

Comment:
is there any way to ensure the validation doesn't mutate the input? I can't think of anything in the type system we could use but it could be a neat constraint to have.  in C++ I'd do something with const and not returning, ie just having read access to the thing under validation.

Reply:
I'm assuming you mean the functions in the validation crate:

I just double checked, and only the functions starting with `required` require ownership (`valid_json` does too, but that is a mistake and unnecessary). On the `required` functions, it makes for a nicer pattern for the caller as it will either return an error or unwrap the `Option`, but it is only designed this way for convenience, and can be changed.  

We may also be able to restrict the generic functions further to prevent passing ownership.